### DISCUSSION TITLE
Asynchronous multi-zone volumes

### DISCUSSION THREAD

**Discussion by amirrezawh** [longhorn/longhorn#3593] [Q&A]
Hello,
I'm looking for a solution for my Kubernetes storage needs. Imagine that I have a Kubernetes cluster with some nodes that are located in different zones. My storage needs to use local volumes on my nodes (for speed) and sync volumes between nodes in an async manner. Is this possible with Longhorn? Would Longhorn work in this scenario?

**Comment by joshimoo** [comment]
We only do sync io between our replicas, so a single io write needs to wait for all replicas to have processed the io.
For local-path you can use https://github.com/rancher/local-path-provisioner which ships with k3s out of the box.
But it does not have sync between the nodes, since it's local-path only.

If you want to elaborate your use case, we might be able to provide additional advice :)

**Reply by amirrezawh** [reply]
Thanks,
I know rancher's provisioner, but I find it so simple.
As I stated in my question, I am interested in implementing a multizone Kubernetes cluster. However, I am unsure of how to handle storage for my cluster. Since I do not use remote storage, I decided to use Longhorn, and I do not use any cloud provider. My nodes should be in different regions, so they need sync volumes, but I think there may be a lot of latency in different zones. Therefore, using sync IO instead of async might affect my cluster's performance.
